pkg/greenlake/login.go

```diff
@@ -0,0 +1,69 @@
+// (C) Copyright 2019 Hewlett Packard Enterprise Development LP.
+
+package greenlake
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/HewlettPackard/hpecli/pkg/logger"
+	"github.com/spf13/cobra"
+)
+
+var glLoginData struct {
+	host,
+	userID,
+	secretKey,
+	tenantID,
+	grantType string
+}
+
+func init() {
+	glLoginCmd.Flags().StringVar(&glLoginData.host, "host", "", "greenlake host/ip address")
+	glLoginCmd.Flags().StringVarP(&glLoginData.userID, "userid", "u", "", "greenlake userid")
+	glLoginCmd.Flags().StringVarP(&glLoginData.secretKey, "secretkey", "s", "", "greenlake secretkey")
+	glLoginCmd.Flags().StringVarP(&glLoginData.tenantID, "tenantid", "t", "", "greenlake tenantid")
+	_ = glLoginCmd.MarkFlagRequired("host")
+	_ = glLoginCmd.MarkFlagRequired("userid")
+	_ = glLoginCmd.MarkFlagRequired("secretkey")
+	_ = glLoginCmd.MarkFlagRequired("tenantid")
+}
+
+// glLoginCmd represents the green lake login command
+var glLoginCmd = &cobra.Command{
+	Use:   "login",
+	Short: "Login to greenlake: hpecli greenlake login",
+	RunE:  runGLLogin,
+}
+
+func runGLLogin(_ *cobra.Command, _ []string) error {
+	logger.Info("greenlake/login called")
+
+	if !strings.HasPrefix(glLoginData.host, "http") {
+		glLoginData.host = fmt.Sprintf("http://%s", glLoginData.host)
+	}
+
+	logger.Debug("Attempting login with user: %v, at: %v", glLoginData.userID, glLoginData.host)
+	glc := NewGreenLakeClient("client_credentials", glLoginData.userID,
```

Comment:
might be simpler if you just pass in glLoginData as a struct rather than individual elements?
The pull the fields apart in the NewGreenLakeClient function